src/cs_main.h

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2023 The Bitcoin Core developers
+// Copyright (c) 2024 Chaintope Inc.
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_CS_MAIN_H
+#define BITCOIN_CS_MAIN_H
+
+#include <sync.h>
+
+/**
+ * Mutex to guard access to validation specific variables, such as reading
+ * or changing the chainstate.
+ *
+ * This may also need to be locked when updating the transaction pool, e.g. on
+ * AcceptToMemoryPool. See CTxMemPool::cs comment for details.
+ *
+ * The transaction pool has a separate lock to allow reading from it and the
+ * chainstate at the same time.
+ */
+extern RecursiveMutex cs_main;
+extern RecursiveMutex cs_LastBlockFile;
```

Comment:
In Bitcoin, it seems there is only cs_main, but what is cs_LastBlockFile for?

Reply:
This is another global mutex which is shared between multiple files. So it is needed in a common place. In bitcoin, this is moved inside a new class called block manager.